File: examples/Ruby/webmqx-server.rb
#!/usr/bin/env ruby
# encoding: utf-8

require "bunny"

# Connect to RabbitMQ server.
conn = Bunny.new("amqp://guest:guest@127.0.0.1", :automatically_recover => false)
conn.start

ch   = conn.create_channel

class WebmqxServer

  def initialize(ch)
    @ch = ch
  end

  def start()
    @x = @ch.exchange("webmqx", :type => "webmqx")
    @q = @ch.temporary_queue()
	@q.bind(@x, :routing_key => "/ruby-test/1")
	@q.bind(@x, :routing_key => "/ruby-test/1/2")
	@q.bind(@x, :routing_key => "/ruby-test/1/2/3")
	@q.bind(@x, :routing_key => "/ruby-test/3/2/1")

    @q.subscribe(:block => true) do |delivery_info, properties, payload|
      r = self.class.handle(payload)

      puts " [.] respose (#{r})"

      @x.publish(r, :routing_key => properties.reply_to, :correlation_id => properties.correlation_id)
    end
  end


  def self.handle(n)
	  "HelloWorld"
  end
end

begin
  server = WebmqxServer.new(ch)
  puts " [x] Awaiting HTTP requests"
  server.start()
rescue Interrupt => _
  ch.close
  conn.close
  exit(0)
end

File: examples/Javascript-nodejs/webmqx-server.js
#!/usr/bin/env node

var amqp = require('amqplib/callback_api');

amqp.connect('amqp://localhost', function(err, conn) {
  conn.createChannel(function(err, ch) {
	var ex = 'webmqx';

	ch.assertQueue('', {exclusive: true, autoDelete: true}, function(err, q) {
		console.log(" [*] Waiting for http requests in %s. To exit press CTRL+C", q.queue);
		ch.bindQueue(q.queue, ex, '/nodejs-test/1');
		ch.bindQueue(q.queue, ex, '/nodejs-test/1/2');
		ch.bindQueue(q.queue, ex, '/nodejs-test/1/2/3');
		ch.bindQueue(q.queue, ex, '/nodejs-test/3/2/1');

		ch.prefetch(1);
		console.log(' [x] Awaiting RPC requests');
		ch.consume(q.queue, function (msg) {
			//var n = parseInt(msg.content.toString());

			console.log(" [.] %s", msg.content.toString());

			var response = handle();

			var response_body = {  
				"headers": {"content-type":"text/html"}, 
				"body": response   
			}; 
			 
			var response_str = JSON.stringify(response_body); 

			ch.sendToQueue(msg.properties.replyTo,
							new Buffer(response_str),
							{correlationId: msg.properties.correlationId});

			ch.ack(msg);
		});
    });
  });
});

function handle() {
	return 'HelloWorld';
}
